File: app/src/main/java/com/nearsoft/nearsoftjobs/adapter/JobAdapter.java
package com.nearsoft.nearsoftjobs.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.nearsoft.nearsoftjobs.R;
import com.nearsoft.nearsoftjobs.model.Job;

import java.util.List;

/**
 * Created by epool on 7/14/16.
 */
public class JobAdapter extends RecyclerView.Adapter<JobAdapter.JobViewHolder> {

    private List<Job> mJobs;
    private OnJobClickListener mOnJobClickListener;

    public JobAdapter(List<Job> jobs, OnJobClickListener onJobClickListener) {
        mJobs = jobs;
        mOnJobClickListener = onJobClickListener;
    }

    @Override
    public JobViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_job, parent, false);
        return new JobViewHolder(view);
    }

    @Override
    public void onBindViewHolder(JobViewHolder viewHolder, int position) {
        Job job = mJobs.get(position);
        viewHolder.bindView(job);
    }

    @Override
    public int getItemCount() {
        return mJobs.size();
    }

    public interface OnJobClickListener {
        void onJobClicked(Job job);
    }

    class JobViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        private final TextView mJobNameTextView;

        JobViewHolder(View itemView) {
            super(itemView);
            itemView.setOnClickListener(this);
            mJobNameTextView = (TextView) itemView.findViewById(R.id.jobNameTextView);
        }

        void bindView(Job job) {
            mJobNameTextView.setText(job.getName());
        }

        @Override
        public void onClick(View view) {
            mOnJobClickListener.onJobClicked(mJobs.get(getAdapterPosition()));
        }
    }

}

File: app/src/main/java/com/nearsoft/nearsoftjobs/adapter/JobAdapter.kt
package com.nearsoft.nearsoftjobs.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.nearsoft.nearsoftjobs.R
import com.nearsoft.nearsoftjobs.model.Job

/**
 * Created by epool on 7/14/16.
 */
class JobAdapter(private val mJobs: List<Job>, private val mOnJobClickListener: (job: Job) -> Unit) : RecyclerView.Adapter<JobAdapter.JobViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): JobViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_job, parent, false)
        return JobViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: JobViewHolder, position: Int) {
        val job = mJobs[position]
        viewHolder.bindView(job)
    }

    override fun getItemCount(): Int {
        return mJobs.size
    }

    inner class JobViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        private val mJobNameTextView: TextView

        init {
            itemView.setOnClickListener(this)
            mJobNameTextView = itemView.findViewById(R.id.jobNameTextView) as TextView
        }

        fun bindView(job: Job) {
            mJobNameTextView.text = job.name
        }

        override fun onClick(view: View) {
            mOnJobClickListener(mJobs[adapterPosition])
        }

    }

}
